run_gapbs.sh

```diff
@@ -0,0 +1,182 @@
+    
+    # Increase stack size
+    ulimit -s unlimited || warn "Failed to set unlimited stack size"
+    
+    log "Environment configured."
+}
+
+# Run benchmarks
+run_benchmarks() {
+    local benchmarks=("bfs" "pr" "cc" "bc" "sssp" "tc")
+    local graph_types=("rmat" "uniform")
+    
+    log "Starting benchmark suite..."
+    
+    for size in "${GRAPH_SIZES[@]}"; do
+        log "Testing graphs of size 2^$size vertices..."
+        
+        for type in "${graph_types[@]}"; do
+            log "Generating $type graph..."
+            
+            # Check if generator exists
+            if [ ! -f "./generator" ]; then
+                error "Generator executable not found. Please ensure it is compiled."
+            fi
+            
+            if [ "$type" == "rmat" ]; then
+                ./generator -g "$size" -s "$((size+4))" || error "Failed to generate RMAT graph"
+            else
+                ./generator -g "$size" -s "$((size+4))" -r || error "Failed to generate uniform random graph"
+            fi
+            
+            for bench in "${benchmarks[@]}"; do
+                log "Running $bench benchmark..."
+                
+                # Run benchmark with timing
+                TIMEFORMAT="%R"
+                runtime=$( { time ./"$bench" -g "$size" -n "$NUM_ITERATIONS" -v; } 2>&1 )
+                
+                # Log results
+                echo "Benchmark: $bench, Graph: $type, Size: 2^$size, Runtime: ${runtime}s" >> "$LOG_FILE"
+            done
+        done
+    done
+}
+
+# Generate report
+generate_report() {
+    log "Generating benchmark report..."
+    
+    echo -e "\nBenchmark Summary Report" > "report_$(date +%Y%m%d_%H%M%S).txt"
+    echo "=========================" >> "report_$(date +%Y%m%d_%H%M%S).txt"
+    echo "System Information:" >> "report_$(date +%Y%m%d_%H%M%S).txt"
+    echo "CPU: $(lscpu | grep "Model name" | sed 's/Model name: *//')" >> "report_$(date +%Y%m%d_%H%M%S).txt"
+    echo "Memory: $(free -h | awk '/^Mem:/ {print $2}')" >> "report_$(date +%Y%m%d_%H%M%S).txt"
+    echo "Threads: $NUM_THREADS" >> "report_$(date +%Y%m%d_%H%M%S).txt"
+    echo -e "\nBenchmark Results:" >> "report_$(date +%Y%m%d_%H%M%S).txt"
+    grep "Benchmark:" "$LOG_FILE" >> "report_$(date +%Y%m%d_%H%M%S).txt"
+    
+    log "Report generated."
+}
+
+# Cleanup function
+cleanup() {
+    log "Cleaning up..."
+    cd "$INSTALL_DIR" || return
+    make clean
+    rm -f benchmark.graph*
+    log "Cleanup completed."
+}
+
+# Main execution
+main() {
+    echo "GAP Benchmark Suite Runner"
+    echo "========================="
+    
+    check_requirements
+    install_gapbs
+    setup_environment
+    run_benchmarks
+    generate_report
+    cleanup
+    
+    log "Benchmark suite completed successfully!"
+    log "Results are available in $LOG_FILE"
+    log "Detailed report available in report_$(date +%Y%m%d_%H%M%S).txt"
+}
+
+# Trap Ctrl+C and call cleanup
+trap cleanup INT
+
+# Run main function
+main
```

Comment:
it looks like you have not installed any `pre-commit` hooks,
please run `make hooks` and then `make pre-commit`, fixing any issues that appear -- although that command will fix some of the simpler ones like adding a newline at the end of every file, in those cases simply do `git add . && make pre-commit`